manager/assets/modext/widgets/source/modx.panel.sources.js

```diff
@@ -63,221 +68,377 @@ Ext.reg('modx-panel-sources',MODx.panel.Sources);
+                        }
+                    },
+                    scope: this
+                }
+            }
+        }, '->', {
+            xtype: 'textfield',
+            name: 'search',
+            id: 'modx-source-search',
+            cls: 'x-form-filter',
+            emptyText: _('search_ellipsis'),
+            value: MODx.request.query,
+            listeners: {
+                change: {
+                    fn: function(cb, rec, ri) {
+                        this.sourceSearch(cb, rec, ri);
+                    },
+                    scope: this
+                },
+                afterrender: {
+                    fn: function(cb) {
                         if (MODx.request.query) {
                             this.sourceSearch(cb, cb.value);
                             MODx.request.query = '';
                         }
-                    }
-                    ,scope: this
-                }
-                ,'render': {
+                    },
+                    scope: this
+                },
+                render: {
                     fn: function(cmp) {
                         new Ext.KeyMap(cmp.getEl(), {
-                            key: Ext.EventObject.ENTER
-                            ,fn: this.blur
-                            ,scope: cmp
+                            key: Ext.EventObject.ENTER,
+                            fn: this.blur,
+                            scope: cmp
                         });
-                    }
-                    ,scope: this
+                    },
+                    scope: this
                 }
             }
-        },{
-            xtype: 'button'
-            ,text: _('filter_clear')
-            ,id: 'modx-filter-clear'
-            ,cls: 'x-form-filter-clear'
-            ,listeners: {
-                'click': {fn: this.clearFilter, scope: this},
-                'mouseout': { fn: function(evt){
-                    this.removeClass('x-btn-focus');
-                }
+        }, {
+            xtype: 'button',
+            text: _('filter_clear'),
+            id: 'modx-filter-clear',
+            cls: 'x-form-filter-clear',
+            listeners: {
+                click: {
+                    fn: this.clearFilter,
+                    scope: this
+                },
+                mouseout: {
+                    fn: function(evt) {
+                        this.removeClass('x-btn-focus');
+                    }
                 }
             }
-        }]
+        }],
+        viewConfig: {
+            forceFit: true,
+            scrollOffset: 0,
+            getRowClass: function(record, index, rowParams, store) {
+                // Adds the returned class to the row container's css classes
+                const disableClasses = record.json.isProtected ? 'modx-protected-row disable-selection' : 'disable-selection' ;
+                return !this.grid.userCanDeleteRecord(record) ? disableClasses : '';
+            }
```

Comment:
+ more extensible code
+ no duplication

```suggestion
            getRowClass: function(record, index, rowParams, store) {
                // Adds the returned class to the row container's css classes
                if (this.grid.userCanDeleteRecord(record)) {
                    return ''
                }
                const disableClasses = ['disable-selection']
                if (record.json.isProtected) {
                    disableClasses.push('modx-protected-row')
                }
                return disableClasses.join(' ');
            }
```

Reply:
OK, I updated this in a little different way to avoid duplication (see below). Because this will only be implemented in a handful of places, has to be done on each class where it is implemented, and it's unlikely the class lists for each will need to be added to, I don't think extensibility is really a factor.
```
getRowClass: function(record, index, rowParams, store) {
    // Adds the returned class to the row container's css classes
    if (this.grid.userCanDeleteRecord(record)) {
        return '';
    }
    const rowClasses = 'disable-selection';
    return record.json.isProtected ? `modx-protected-row  ${rowClasses}` : rowClasses ;
}
```